Automattic/block-experiments
# Add IE prefixes to layout grid

This uses autoprefixer in conjunction with its [`no-autoplace`](https://github.com/postcss/autoprefixer#control-comments) feature to add IE11/Edge specific prefixes to the grid CSS.

The prefixed CSS does not appear to affect other browsers.

Note that IE11 doesn't support certain grid features, even with prefixes. Specifically `grid-gap` is not available and so the columns have no gaps between them. We could probably fake it with margins/padding in some way, but I'm not sure if it's worth it.

I believe Edge supports `grid-gap`, but since it's now turned into Chrome I've been unable to test.

## Testing

1) Create a post with a grid
2) Without this PR applied, view the post and 'save page as' a complete web page.
3) Start IE, open inspector, and ensure emulation document mode is set to IE11
4) Drop the saved HTML file into IE and confirm that the grid is displayed vertically

![before](https://user-images.githubusercontent.com/1277682/75982411-85471400-5ede-11ea-975c-a2a0f7148a5c.jpg)

5) Repeat but with this PR applied. Confirm that the grid is now horizontal
![after](https://user-images.githubusercontent.com/1277682/75982417-88da9b00-5ede-11ea-93f0-07deecec6ff1.jpg)

Fixes #26
